src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -184,39 +186,143 @@ public async Task EmissionsForecastsBatch_MissingRequiredParams_ReturnsBadReques
         Assert.That(result?.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
     }
 
-    [Test]
-    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk()
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T09:00:00Z", "2021-09-01T12:00:00Z", "eastus", 1, TestName = "EmissionsForecastsBatch expects OK for eastus 1 element")]
+    [TestCase("2021-09-01T08:30:00Z", "2021-09-01T08:30:00Z", "2021-09-02T08:30:00Z", "westus", 3, TestName = "EmissionsForecastsBatch expects OK for westus 3 element")]
+    public async Task EmissionsForecastsBatch_SupportedDataSources_ReturnsOk(DateTimeOffset reqAt, DateTimeOffset start, DateTimeOffset end, string location, int nelems)
     {
         IgnoreTestForDataSource("data source does not implement '/emissions/forecasts/batch'", DataSourceType.JSON);
 
         _dataSourceMocker.SetupBatchForecastMock();
+        var inputData = Enumerable.Range(0, nelems).Select(x => new EmissionsForecastBatchDTO() 
+        {
+            RequestedAt = reqAt,
+            DataStartAt = start,
+            DataEndAt = end,
+            Location = location
+        });
+
+        using (var result = await PostJSONBodyToURI(inputData, batchForecastURI))
+        {
+            Assert.That(result, Is.Not.Null);
+            Assert.That(result!.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(result!.Content, Is.Not.Null);
+            using (var data = await result!.Content.ReadAsStreamAsync())
+            {
+                Assert.That(data, Is.Not.Null);
+                var forecasts = await JsonSerializer.DeserializeAsync<IEnumerable<EmissionsForecastDTO>>(data);
+                Assert.That(forecasts, Is.Not.Null);
+                Assert.That(forecasts!.Count, Is.EqualTo(inputData.Count()));
+                foreach (var forecast in forecasts!)
+                {
+                    Assert.That(forecast.Location, Is.EqualTo(location));
+                    Assert.That(forecast.DataStartAt, Is.EqualTo(start));
+                    Assert.That(forecast.DataEndAt, Is.EqualTo(end));
+                    Assert.That(forecast.RequestedAt, Is.EqualTo(reqAt));
+                    Assert.That(forecast.GeneratedAt, Is.Not.Null);
+                    Assert.That(forecast.OptimalDataPoint, Is.Not.Null);
+                    Assert.That(forecast.ForecastData, Is.Not.Null);
+                }
+            }
+        }
+    }
 
-        var inputData = new List<EmissionsForecastBatchDTO>()
+    [TestCase("2022-1-1T04:05:06Z", "2022-1-2T04:05:06Z", "eastus", TestName = "EmissionsActual expects OK for eastus")]
+    [TestCase("2021-12-25", "2021-12-26", "westus", TestName = "EmissionsActual expects OK for westus")]
```

Comment:
Similar questions to the above test:
- Can you provide details around the value of second test?
- Prefer string dates in the method signature
- remove content checking except for `Is.Not.Null`